presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -226,6 +249,25 @@ else if (writerTarget instanceof ExecutionWriterTarget.InsertHandle) {
         }
     }
 
+    private static void commit(Session session, TransactionManager transactionManager)
```

Comment:
curious: why make this method to be static and always pass in `this.session` and `this.transactionManager`? (compared with the previous solution to have instance method?

Reply:
It has to be used inside the `PrestoSparkQueryExecution` that is static. I don't want to make it non static to avoid implicit references.